### PR TITLE
Add `com.iexec.commons.poco.order` package

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

File: src/main/java/com/iexec/commons/poco/order/RequestOrder.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                Objects.equals(datasetmaxprice, that.datasetmaxprice) &&
+                Objects.equals(workerpool, that.workerpool) &&
+                Objects.equals(workerpoolmaxprice, that.workerpoolmaxprice) &&
+                Objects.equals(volume, that.volume) &&
+                Objects.equals(category, that.category) &&
+                Objects.equals(trust, that.trust) &&
+                Objects.equals(tag, that.tag) &&
+                Objects.equals(beneficiary, that.beneficiary) &&
+                Objects.equals(callback, that.callback) &&
+                //Objects.equals(params, that.params) &&
```
Comment (mcornaton) [inline src/main/java/com/iexec/commons/poco/order/RequestOrder.java:99]: To be consistent, I'd prefer:
1. Either remove that commented out line
2. Add commented out lines for `salt` and `sign`

File: src/test/java/com/iexec/commons/poco/order/DatasetOrderTests.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        assertThat(web3jDatasetOrder.datasetprice).isEqualTo(datasetOrder.getDatasetprice());
+        assertThat(web3jDatasetOrder.volume).isEqualTo(datasetOrder.getVolume());
+        assertThat(web3jDatasetOrder.tag).isEqualTo(Numeric.hexStringToByteArray(datasetOrder.getTag()));
+        assertThat(web3jDatasetOrder.apprestrict).isEqualTo(datasetOrder.getApprestrict());
+        assertThat(web3jDatasetOrder.workerpoolrestrict).isEqualTo(datasetOrder.getWorkerpoolrestrict());
+        assertThat(web3jDatasetOrder.requesterrestrict).isEqualTo(datasetOrder.getRequesterrestrict());
+        assertThat(web3jDatasetOrder.salt).isEqualTo(Numeric.hexStringToByteArray(datasetOrder.getSalt()));
+    }
+
+}
```
Comment (mcornaton) [inline src/test/java/com/iexec/commons/poco/order/DatasetOrderTests.java:70]: Missing new line:
```suggestion
}

```

File: src/main/java/com/iexec/commons/poco/order/RequestOrder.java (hunk 1 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/commons/poco/order/RequestOrder.java]: Please see 3534b78 for method removal

**Review by mcornaton — APPROVED** [review]
Thanks

**Pull request merged**